rp2040-hal/src/spi.rs

```diff
@@ -263,22 +285,26 @@ impl<D: SpiDevice, P: ValidSpiPinout<D>, const DS: u8> Spi<Disabled, D, P, DS> {
     }
 
     /// Initialize the SPI in master mode
-    pub fn init<F: Into<HertzU32>, B: Into<HertzU32>, M: Into<Mode>>(
+    pub fn init<F: Into<HertzU32>, B: Into<HertzU32>>(
         self,
         resets: &mut RESETS,
         peri_frequency: F,
         baudrate: B,
-        mode: M,
+        frame_format: FrameFormat,
     ) -> Spi<Enabled, D, P, DS> {
-        self.init_spi(resets, peri_frequency, baudrate, mode.into(), false)
+        self.init_spi(resets, peri_frequency, baudrate, frame_format, false)
     }
 
     /// Initialize the SPI in slave mode
-    pub fn init_slave<M: Into<Mode>>(self, resets: &mut RESETS, mode: M) -> Spi<Enabled, D, P, DS> {
+    pub fn init_slave(
+        self,
+        resets: &mut RESETS,
+        frame_format: FrameFormat,
+    ) -> Spi<Enabled, D, P, DS> {
         // Use dummy values for frequency and baudrate.
         // With both values 0, set_baudrate will set prescale == u8::MAX, which will break if debug assertions are enabled.
         // u8::MAX is outside the allowed range 2..=254 for CPSDVSR, which might interfere with proper operation in slave mode.
-        self.init_spi(resets, 1000u32.Hz(), 1000u32.Hz(), mode.into(), true)
+        self.init_spi(resets, 1000u32.Hz(), 1000u32.Hz(), frame_format, true)
```

Comment:
```suggestion
        self.init_spi(resets, 1000u32.Hz(), 1000u32.Hz(), frame_format.into(), true)
```

Reply:
Is this API really better? Personally, I don't like those `.into()` conversions.

Sure, they are sometimes necessary, because rust prefers to be explicit about type conversions, and that's good. But if we could avoid them in the common case, it'd be even better.

Or, from a different perspective: Are there other usages of the API that become nicer with your suggestion?

Reply:
Well, my suggestion removes the manual wrapping and the `.into()` that this PR introduces.

So, IMHO:
```rust
    let mut spi = spi.init(
        &mut pac.RESETS,
        clocks.peripheral_clock.freq(),
        16.MHz(),
        embedded_hal::spi::MODE_0,
    );
```
is better/more readable than
```rust
    let mut spi = spi.init(
        &mut pac.RESETS,
        clocks.peripheral_clock.freq(),
        16.MHz(),
        hal::spi::FrameFormat::MotorolaSpi(embedded_hal::spi::MODE_0.into()),
    );
```

Reply:
Good point, the modes from EH are probably much more commonly used than other two, so being able to use them without explicitly specifying Motorola frame format is nice.

Reply:
I think it could make sense to implement the conversions from the mode structs embedded_hal crate, but only because they do not apply to the other frame formats. It might not be clear from a usage standpoint why this is the case.

Perhaps the inconvenience factor might instead be begging the question of why we are intent on instantiating this by converting from embedded_hal::spi::MODE_0 every time.

I'm wondering if an alternate approach of making it easier to get the default value by providing a constant / implementing Default would be acceptable. Users could very easily see that there is a default format and that that format is MotorolaSpi with the usual values of phase and polarity, as opposed to having to understand that there is conversion magic from the embedded_hal structs and why it ends up being MotorolaSpi format when it does.

```rust
// not necessarily what I'm saying it should be named, but something like 
    let mut spi = spi.init(
        &mut pac.RESETS,
        clocks.peripheral_clock.freq(),
        16.MHz(),
        hal::spi::DEFAULT_FRAME_FORMAT,
    );
    // or
   let mut spi = spi.init(
        &mut pac.RESETS,
        clocks.peripheral_clock.freq(),
        16.MHz(),
        hal::spi::FrameFormat::default(),
    );
    // or
    let mut spi = spi.init(
        &mut pac.RESETS,
        clocks.peripheral_clock.freq(),
        16.MHz(),
        Default::default(),
    );
```

What do you think?

Reply:
I don't think there is one mode which could be considered "usual values of phase and polarity". And getting this wrong can be difficult to debug. So I'd prefer to select the desired mode explicitly.